backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt

```diff
@@ -209,7 +209,17 @@ internal val psiToIrPhase = konanUnitPhase(
                     moduleDescriptor, symbolTable,
                     config.configuration.languageVersionSettings
             )
-            val irProviderForCEnumsAndCStructs = IrProviderForCEnumAndCStructStubs(generatorContext, interopBuiltIns, symbols)
+
+            val generationMode = if (llvmModuleSpecification.isFinal) {
+                IrProviderForCEnumAndCStructStubs.GenerationMode.WholeWorld
```

Comment:
Is this true? Final module can consume caches.

Reply:
Currently, yes. IIUC, `LlvmModuleSpecificationImpl.containsDeclaration` will return false for declarations from cached libraries, so it works correctly.
We can generate lazy IR for declarations from cached libraries, but it will be an optimization that should be implemented separately (this PR is bug fix).

Reply:
> We can generate lazy IR for declarations from cached libraries

If this code generate non-lazy IR for cached libraries, how does it avoid placing it to current module? If it doesn't, then it can end up with same declaration in multiple object files.